### ISSUE TITLE
aria-activedescendant is not working properly in Select

### ISSUE THREAD

**Issue by hardikmodi1** [uber/baseweb#3880]
While using SelectBox, aria-activedescendant property is not working correctly. Here is the [Codesandbox link](https://codesandbox.io/s/base-web-select-forked-uifbu) 

Currently it gives aria-activedescendant property a id of the selected option from the list. Due to this screen reader is not able to pronounce the option while we are navigating the list using arrow key.

## Expected Behavior
According to [this a11y criteria](https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html), it should have id of the option that is currently focused

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.96.0  |
| React   |   16.8.0      |
| browser |     Chrome    |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.
